src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -48,6 +52,23 @@ cdef class RPCState:
     cdef tuple invocation_metadata(self):
         return _metadata(&self.request_metadata)
 
+    cdef void raise_for_termination(self) except *:
+        """Raise exceptions if RPC is not running.
+
+        Server method handlers may suppress the abort exception. We need to halt
+        the RPC execution in that case. This function needs to be called after
+        running application code.
+
+        Also, the server may stop unexpected. We need to check before calling
+        into Core functions, otherwise, segfault.
+        """
+        if self.abort_exception is not None:
+            raise self.abort_exception
+        if self.status_sent:
+            raise RuntimeError(_RPC_FINISHED_DETAILS)
```

Comment:
Optional: It might be nice for us to have a common base class for our exceptions. I can imagine a user of the library wanting to do the following:
```python
try:
  do_grpc_stuff()
except grpc.BaseException as e:
  logging.warning("Uh-oh. gRPC failed but not my application: %s", e)
  degrade_gracefully()
```

Reply:
That is a good idea to me. I noted in #21624. Will do in separate PR.